mmv1/products/tpuv2/QueuedResource.yaml

```
@@ -0,0 +1,110 @@
+# Copyright 2024 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+name: 'QueuedResource'
+description: |
+  A Cloud TPU Queued Resource.
+min_version: 'beta'
+references:
+  guides:
+    'Official Documentation': 'https://cloud.google.com/tpu/docs/'
+  api: 'https://cloud.google.com/tpu/docs/reference/rest/v2/projects.locations.queuedResources'
+base_url: 'projects/{{project}}/locations/{{zone}}/queuedResources'
+self_link: 'projects/{{project}}/locations/{{zone}}/queuedResources/{{name}}'
+create_url: 'projects/{{project}}/locations/{{zone}}/queuedResources?queuedResourceId={{name}}'
+update_url: 'projects/{{project}}/locations/{{zone}}/queuedResources/{{name}}'
+update_verb: 'PATCH'
+update_mask: true
+autogen_async: true
+async:
+  actions: ['create', 'delete']
+  type: 'OpAsync'
+  operation:
+    base_url: '{{op_id}}'
+    path: 'name'
+    wait_ms: 1000
+  result:
+    path: 'response'
+    resource_inside_response: true
+  status:
+    path: 'done'
+    complete: true
+    allowed:
+      - true
+      - false
+  error:
+    path: 'error'
+    message: 'message'
+examples:
+  - name: 'tpu_v2_queued_resource_basic'
+    primary_resource_id: 'qr'
+    min_version: 'beta'
+    vars:
+      qr_name: 'test-qr'
+      tpu_name: 'test-tpu'
+    test_env_vars:
+      project: 'PROJECT_NAME'
+    skip_vcr: true
+parameters:
+  - name: 'zone'
+    type: String
+    description: |
+      The GCP location for the Queued Resource. If it is not provided, the provider zone is used.
+    url_param_only: true
+    immutable: true
+    default_from_api: true
+properties:
+  - name: 'name'
+    type: String
+    description: |
+      The immutable name of the Queued Resource.
+    required: true
+    immutable: true
+    custom_flatten: 'templates/terraform/custom_flatten/name_from_self_link.tmpl'
+  - name: 'tpu'
+    type: NestedObject
+    description: |
+      Defines a TPU resource.
+    properties:
+      - name: 'nodeSpec'
+        type: Array
+        description: |
+          The TPU node(s) being requested.
+        item_type:
+          type: NestedObject
+          properties:
+            - name: 'parent'
+              description: |
+                The parent resource name.
+              required: true
```

We need diff suppression on this field because users can use the project id in the value and the API will return the value containing the project number. Without diff suppression users will see unexpected diffs.

```suggestion
              required: true
              diff_suppress_func: 'tpgresource.ProjectNumberDiffSuppress'
```

---

Updated to use this `diff_suppress_func`